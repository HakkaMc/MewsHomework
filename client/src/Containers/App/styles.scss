.app {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}

.container {
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
}

.paper {
  flex: 1 0 auto;
  padding: 20px 40px;
  margin-bottom: 60px;
}
