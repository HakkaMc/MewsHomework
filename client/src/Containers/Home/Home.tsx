import React from "react";
import { Typography } from "@material-ui/core";
import {
	FormattedMessage,
	injectIntl,
	WrappedComponentProps as IntlProps
} from "react-intl";
import { compose } from "recompose";

import { Search } from "../../Components";
import { messages } from "../../constants/locales/messages";

export type Props = IntlProps;

export const Home: React.FC<Props> = (props: Props) => {
	return (
		<>
			<Typography variant="h6">
				<FormattedMessage {...messages.Home.title} />
			</Typography>
			<Search
				placeholder={props.intl.formatMessage(messages.Home.placeholder)}
			/>
		</>
	);
};

export default compose<Props, any>(injectIntl)(Home);
