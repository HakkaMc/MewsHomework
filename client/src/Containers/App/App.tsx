import React from "react";
import { Container, Paper } from "@material-ui/core";
import { connect } from "react-redux";
import { compose } from "recompose";

import { ReduxState } from "../../reducers";
import Header from "../Header/Header";
import Home from "../Home/Home";

import styles from "./styles.scss";

// Example of component lazy loading
const Movies = React.lazy(() => import("../Movies/Movies"));

export interface Props {
	totalResults: number;
}

export const App: React.FC<Props> = (props: Props) => (
	<div className={styles.app}>
		<Header />

		<Container className={styles.container}>
			<Paper elevation={3} className={styles.paper}>
				<React.Suspense fallback={<div>Loading...</div>}>
					{!props.totalResults && <Home />}
					{!!props.totalResults && <Movies />}
				</React.Suspense>
			</Paper>
		</Container>
	</div>
);

export default compose<Props, any>(
	connect(
		(state: ReduxState) => ({
			totalResults: state.movies.totalResults
		}),
		null
	)
)(App);
