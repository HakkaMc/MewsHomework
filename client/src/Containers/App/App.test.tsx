import React from "react";
import { shallow } from "enzyme";

import { App, Props } from "./App";
import Home from "../Home/Home";

describe("<App/>", () => {
	let props: Props;

	beforeEach(() => {
		props = {
			totalResults: 0
		};
	});

	it("Render with totalResults: 0", () => {
		props.totalResults = 0;
		const wrapper = shallow(<App {...props} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find(Home)).toHaveLength(1);
	});

	it("Render with totalResults: 1", () => {
		props.totalResults = 1;
		const wrapper = shallow(<App {...props} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find(Home)).toHaveLength(0);
	});
});
