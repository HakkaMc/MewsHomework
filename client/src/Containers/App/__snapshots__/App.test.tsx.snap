// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Render with totalResults: 0 1`] = `
<div>
  <injectIntl(Connect(Header)) />
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Paper))
      elevation={3}
    >
      <Suspense
        fallback={
          <div>
            Loading...
          </div>
        }
      >
        <injectIntl(Home) />
      </Suspense>
    </WithStyles(ForwardRef(Paper))>
  </WithStyles(ForwardRef(Container))>
</div>
`;

exports[`<App/> Render with totalResults: 1 1`] = `
<div>
  <injectIntl(Connect(Header)) />
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Paper))
      elevation={3}
    >
      <Suspense
        fallback={
          <div>
            Loading...
          </div>
        }
      >
        <lazy />
      </Suspense>
    </WithStyles(ForwardRef(Paper))>
  </WithStyles(ForwardRef(Container))>
</div>
`;
